.travis.yml

```diff
@@ -0,0 +1,26 @@
+language: cpp
+dist: trusty
+sudo: true
+
+env:
+  matrix:
+    - CONFIG=DEFAULT
+    - CONFIG=ALL_ON
+    - CONFIG=ALL_OFF
+
+addons:
+  apt:
+    packages:
+      - libpcap-dev
+      - libunwind8-dev
+
+install:
+  - git clone --depth=1 --branch=mbedtls-2.5.1 https://github.com/ARMmbed/mbedtls.git $HOME/mbedtls && mkdir -p $HOME/mbedtls-build && pushd $HOME/mbedtls-build && cmake -DCMAKE_INSTALL_PREFIX=/usr/local -DUSE_SHARED_MBEDTLS_LIBRARY=ON -DUSE_STATIC_MBEDTLS_LIBRARY=OFF -DENABLE_PROGRAMS=OFF -DENABLE_TESTING=OFF $HOME/mbedtls && make && sudo make install && popd
+  - git clone --depth=1 --branch=rewrite https://github.com/wiringX/wiringX.git $HOME/wiringx && mkdir -p $HOME/wiringx-build && pushd $HOME/wiringx-build && cmake $HOME/wiringx && make && sudo make install && popd
+
+before_script:
+  - if [[ $CONFIG == "ALL_OFF" ]]; then sed "s/\(set(.*\) ON/\1 OFF/" -i CMakeConfig.txt; fi
+  - if [[ $CONFIG == "ALL_ON" ]]; then sed "s/\(set(.*\) OFF/\1 ON/" -i CMakeConfig.txt; fi
+
+script:
+  - mkdir build && pushd build && cmake .. && make && ./pilight-unittest && sudo make install
```

Comment:
In case of a build error, there will be max. 1 more line from the other make thread being printed before `make` terminates, so I think it's not to bad compared to the overall time saved by using `-j2`

I think `VERBOSE=1` is overkill:

```
/usr/bin/cmake -H/home/git/git/pilight/pilight -B/home/git/git/pilight/pilight/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/git/git/pilight/pilight/build/CMakeFiles /home/git/git/pilight/pilight/build/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight.dir/build.make CMakeFiles/pilight.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight.dir/build.make CMakeFiles/pilight.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[ 91%] Built target pilight
make -f CMakeFiles/pilight_shared.dir/build.make CMakeFiles/pilight_shared.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight_shared.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight_shared.dir/build.make CMakeFiles/pilight_shared.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight_shared.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[ 91%] Built target pilight_shared
make -f CMakeFiles/pilight_static.dir/build.make CMakeFiles/pilight_static.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight_static.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight_static.dir/build.make CMakeFiles/pilight_static.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight_static.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[ 92%] Built target pilight_static
make -f CMakeFiles/pilight-raw.dir/build.make CMakeFiles/pilight-raw.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight-raw.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight-raw.dir/build.make CMakeFiles/pilight-raw.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight-raw.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[ 92%] Built target pilight-raw
make -f CMakeFiles/pilight-daemon.dir/build.make CMakeFiles/pilight-daemon.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight-daemon.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight-daemon.dir/build.make CMakeFiles/pilight-daemon.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight-daemon.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[ 93%] Built target pilight-daemon
make -f CMakeFiles/pilight-uuid.dir/build.make CMakeFiles/pilight-uuid.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight-uuid.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight-uuid.dir/build.make CMakeFiles/pilight-uuid.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight-uuid.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[ 94%] Built target pilight-uuid
make -f CMakeFiles/pilight-flash.dir/build.make CMakeFiles/pilight-flash.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight-flash.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight-flash.dir/build.make CMakeFiles/pilight-flash.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight-flash.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[ 95%] Built target pilight-flash
make -f CMakeFiles/pilight-send.dir/build.make CMakeFiles/pilight-send.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight-send.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight-send.dir/build.make CMakeFiles/pilight-send.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight-send.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[ 96%] Built target pilight-send
make -f CMakeFiles/pilight-control.dir/build.make CMakeFiles/pilight-control.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight-control.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight-control.dir/build.make CMakeFiles/pilight-control.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight-control.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[ 97%] Built target pilight-control
make -f CMakeFiles/pilight-sha256.dir/build.make CMakeFiles/pilight-sha256.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight-sha256.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight-sha256.dir/build.make CMakeFiles/pilight-sha256.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight-sha256.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[ 98%] Built target pilight-sha256
make -f CMakeFiles/pilight-receive.dir/build.make CMakeFiles/pilight-receive.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight-receive.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight-receive.dir/build.make CMakeFiles/pilight-receive.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight-receive.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[ 99%] Built target pilight-receive
make -f CMakeFiles/pilight-debug.dir/build.make CMakeFiles/pilight-debug.dir/depend
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
cd /home/git/git/pilight/pilight/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/git/git/pilight/pilight /home/git/git/pilight/pilight /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build /home/git/git/pilight/pilight/build/CMakeFiles/pilight-debug.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
make -f CMakeFiles/pilight-debug.dir/build.make CMakeFiles/pilight-debug.dir/build
make[2]: Entering directory '/home/git/git/pilight/pilight/build'
make[2]: Nothing to be done for 'CMakeFiles/pilight-debug.dir/build'.
make[2]: Leaving directory '/home/git/git/pilight/pilight/build'
[100%] Built target pilight-debug
make[1]: Leaving directory '/home/git/git/pilight/pilight/build'
/usr/bin/cmake -E cmake_progress_start /home/git/git/pilight/pilight/build/CMakeFiles 0
```

compared to:

```
[ 91%] Built target pilight
[ 91%] Built target pilight_shared
[ 92%] Built target pilight_static
[ 92%] Built target pilight-raw
[ 93%] Built target pilight-daemon
[ 94%] Built target pilight-uuid
[ 95%] Built target pilight-flash
[ 96%] Built target pilight-send
[ 97%] Built target pilight-control
[ 98%] Built target pilight-sha256
[ 99%] Built target pilight-receive
[100%] Built target pilight-debug
```

And that's just on an already built repo.